### PR TITLE
Fix Sonar issues #440

### PR DESCRIPTION
**Pull request by gvdutra**
Still, a long list to be solved, I will be doing it as I have time.

### CODE REVIEW

**Comment by JohannesLichtenberger** [comment]
Can you fix the failing tests?

**Comment by gvdutra** [comment]
yes for sure, so strange I ran locally and it was all good. lemme check it.

**Comment by JohannesLichtenberger** [comment]
Can you revert this path kind change? Meaning the `getNodeKind()` should be right.

File: bundles/sirix-core/src/main/java/org/sirix/access/trx/node/AbstractNodeHashing.java [hunk 1]
```diff
@@ -158,7 +158,7 @@ private void postorderAdd() {
       cursorToRoot = pageTrx.prepareRecordForModification(getCurrentNode().getNodeKey(), IndexType.DOCUMENT, -1);
       hashCodeForParent = getCurrentNode().computeHash().add(hashCodeForParent.multiply(PRIME));
       // Caring about attributes and namespaces if node is an element.
-      if (cursorToRoot.getKind() == NodeKind.ELEMENT) {
+      if (cursorToRoot.getPathKind() == NodeKind.ELEMENT) {
```
Comment (JohannesLichtenberger) [inline bundles/sirix-core/src/main/java/org/sirix/access/trx/node/AbstractNodeHashing.java:161]: this should be the `getKind()`-method, but somewhere I think you changed something between `getKind()` and `getPathKind()`. But over here it checks the type of the node itself, not the path type/kind.

Comment (JohannesLichtenberger) [reply bundles/sirix-core/src/main/java/org/sirix/access/trx/node/AbstractNodeHashing.java:161]: Accordingly all the other changes in this commit :)

Comment (gvdutra) [reply bundles/sirix-core/src/main/java/org/sirix/access/trx/node/AbstractNodeHashing.java:161]: ok, I will revert the getKind and getPathKind that I changed on main/java/org/sirix/index/path/summary/PathNode.java for now, but, just for you know sonar is complaining as a critical bug. https://sonarcloud.io/project/issues?id=sirixdb_sirix&open=AXy4_zUSQgQXZ6uows07&resolved=false&severities=CRITICAL&types=BUG

**Comment by gvdutra** [comment]
Done :) if anything needs to be changed please let me know, after this gets merged, I will continue working on more sonar issues.

File: bundles/sirix-core/src/main/java/org/sirix/access/trx/node/AbstractNodeHashing.java (hunk 1 above)
Comment (gvdutra) [reply bundles/sirix-core/src/main/java/org/sirix/access/trx/node/AbstractNodeHashing.java:161]: Done :) if anything needs to be changed please let me know, after this gets merged, I will continue working on more sonar issues.

Comment (JohannesLichtenberger) [reply bundles/sirix-core/src/main/java/org/sirix/access/trx/node/AbstractNodeHashing.java:161]: getKind() should return NodeKind.PATH as that's the type of the PathNode. The `getPathKind()`-method should return which type of a path node the current node denotes (for JSON: array or object fields, for XML: attribute or element).

That is a path node is a node of a path summary, which simply is a "small" tree, which denotes all paths in a resource :-) Hope that makes sense and helps in getting the context. Did you already push your revert?

Comment (gvdutra) [reply bundles/sirix-core/src/main/java/org/sirix/access/trx/node/AbstractNodeHashing.java:161]: Oh I see, so we might need to add the @ignore tag for sonar on this class and/or might be useful to add a comment on it, as Sonar thinks this is an issue. yes, changes on PathNode class are reverted! :)

Comment (JohannesLichtenberger) [reply bundles/sirix-core/src/main/java/org/sirix/access/trx/node/AbstractNodeHashing.java:161]: Yes, I've closed the issue on Sonar with a short description. We have to be careful about false positives.

I can't find the reverted changes. `AbstractNodeHashing` for instance still uses `getPathKind()` as far as I can see.

Comment (gvdutra) [reply bundles/sirix-core/src/main/java/org/sirix/access/trx/node/AbstractNodeHashing.java]: it got messy I will commit again in another PR.

Comment (JohannesLichtenberger) [reply bundles/sirix-core/src/main/java/org/sirix/access/trx/node/AbstractNodeHashing.java]: Thanks and looking forward to your contributions :-)

**Pull request closed without merging**